spec/system/consumer/split_checkout_tax_incl_spec.rb

```diff
@@ -0,0 +1,226 @@
+  }
+  let!(:free_shipping) {
+    create(:shipping_method, distributors: [distributor], require_ship_address: true,
+                             name: "Delivery", description: "Payment without fee",
+                             calculator: Calculator::FlatRate.new(preferred_amount: 0.00))
+  }
+  let!(:free_payment) {
+    create(:payment_method, distributors: [distributor],
+                            name: "Payment without Fee", description: "Payment without fee",
+                            calculator: Calculator::FlatRate.new(preferred_amount: 0.00))
+  }
+  let!(:order_within_zone) {
+    create(:order, order_cycle: order_cycle, distributor: distributor, user: user_within_zone,
+                   bill_address: address_within_zone, ship_address: address_within_zone,
+                   state: "cart", line_items: [create(:line_item, variant: variant_with_tax)])
+  }
+  let!(:order_outside_zone) {
+    create(:order, order_cycle: order_cycle, distributor: distributor, user: user_outside_zone,
+                   bill_address: address_outside_zone, ship_address: address_outside_zone,
+                   state: "cart", line_items: [create(:line_item, variant: variant_with_tax)])
+  }
+
+  before do
+    # assures tax is charged in dependence of shipping address
+    Spree::Config.set(tax_using_ship_address: true)
+  end
+
+  describe "an included tax" do
+    context "as superadmin" do
+      before { login_as_admin }
+
+      it "checks all tax settings are correctly set" do
+        visit "/admin/zones/#{zone.id}/edit"
+        expect(page).to have_field('zone_default_tax', checked: true)
+
+        visit "/admin/tax_settings/edit"
+        expect(page).to have_field('preferences_products_require_tax_category', checked: false)
+
+        visit "/admin/tax_categories/#{tax_category.id}/edit"
+        expect(page).to have_field('tax_category_is_default', checked: false)
+
+        visit "/admin/tax_rates/#{tax_rate.id}/edit"
+        expect(page).to have_field('tax_rate_included_in_price', checked: true)
+      end
+
+      after { logout }
+    end
+
+    describe "for a customer with shipping address within the tax zone" do
+      context "on legacy checkout" do
+        before do
+          set_order order_within_zone
+          login_as(user_within_zone)
+        end
+
+        it "will be charged tax on the order" do
+          visit checkout_path
+
+          find(:xpath, '//*[@id="shipping"]/ng-form/dd').click
+          choose free_shipping.name.to_s
+
+          within "#payment" do
+            choose free_payment.name.to_s
+          end
+
+          click_on "Place order now"
+
+          # UI checks
+          expect(page).to have_selector('#order_total', text: with_currency(10.00))
+          expect(page).to have_selector('#tax-row', text: with_currency(1.15))
+
+          # DB checks
+          assert_db_tax
+        end
+
+        after { logout }
+      end
+
+      context "on split-checkout" do
```

Comment:
This context block seems to be responsible for breaking the build:

```
(base) ffurtado@ff-LAT:~/openfoodnetwork$ bundle exec rspec --bisect spec/system/consumer/split_checkout_tax_not_incl_spec.rb spec/system/consumer/split_checkout_tax_incl_spec.rb --seed 11
Bisect started using options: "spec/system/consumer/split_checkout_tax_not_incl_spec.rb spec/system/consumer/split_checkout_tax_incl_spec.rb --seed 11"
Running suite to find failures... (2 minutes 10.7 seconds)
Starting bisect with 2 failing examples and 9 non-failing examples.
Checking that failure(s) are order-dependent... failure appears to be order-dependent

Round 1: bisecting over non-failing examples 1-9 .. ignoring examples 6-9 (1 minute 34.13 seconds)
Round 2: bisecting over non-failing examples 1-5 . ignoring examples 1-3 (36.73 seconds)
Round 3: bisecting over non-failing examples 4-5 .. multiple culprits detected - splitting candidates (58.9 seconds)
Bisect complete! Reduced necessary non-failing examples from 9 to 2 in 3 minutes 40 seconds.

The minimal reproduction command is:
  rspec ./spec/system/consumer/split_checkout_tax_incl_spec.rb[1:1:2:2:1] ./spec/system/consumer/split_checkout_tax_not_incl_spec.rb[1:1:1:1,1:1:2:1:1,1:1:2:2:1] --seed 11

```

So, running:

`bundle exec rspec ./spec/system/consumer/split_checkout_tax_incl_spec.rb[1:1:2:2:1] ./spec/system/consumer/split_checkout_tax_not_incl_spec.rb[1:1:1:1,1:1:2:1:1,1:1:2:2:1]
`

Effectively makes the second spec fail, and explains the broken build.

I'm just not sure how to further debug it.